treeherder/perf/management/commands/synthesize_backfill_report.py

```diff
@@ -0,0 +1,105 @@
+from argparse import ArgumentError
+from datetime import (datetime,
+                      timedelta)
+from typing import (List,
+                    Tuple)
+
+from django.core.management.base import BaseCommand
+from django.db.models.query import QuerySet
+
+from treeherder.perf.alerts import (AlertsPicker,
+                                    IdentifyRetriggerables)
+from treeherder.perf.models import (PerformanceAlertSummary,
+                                    PerformanceFramework)
+
+
+class Command(BaseCommand):
+    repos_to_retrigger_on = ['autoland', 'mozilla-inbound', 'mozilla-beta']
+    help = "Select most relevant alerts and identify jobs to retrigger."
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--time-window',
+            action='store',
+            type=int,
+            default=60,
+            help="How far back to look for alerts to retrigger (expressed in minutes)."
+        )
+
+        parser.add_argument(
+            '--frameworks',
+            nargs='+',
+            default=None,
+            help="Defaults to all registered performance frameworks."
+        )
+
+        parser.add_argument(
+            '--repositories',
+            nargs='+',
+            default=Command.repos_to_retrigger_on,
+            help=f"Defaults to {Command.repos_to_retrigger_on}."
+        )
+
+    def handle(self, *args, **options):
+        frameworks, repositories, max_timestamp, days_number = self._parse_args(**options)
+        self._validate_args(frameworks, repositories)
+
+        summaries_to_retrigger = self._fetch_by(
+            self._summaries_since(max_timestamp),
+            frameworks,
+            repositories
+        )
+        json_output = self._identify_retriggerables(summaries_to_retrigger, days_number)
+
+        print(json_output)
+
+    def _parse_args(self, **options) -> Tuple[List, List, datetime, timedelta]:
+        return (options['frameworks'],
+                options['repositories'],
+                datetime.now() - timedelta(minutes=options['time_window']),
+                timedelta(days=1))
+
+    def _validate_args(self, frameworks: List[str], repositories: List[str]):
+        if frameworks:
+            available_frameworks = set(PerformanceFramework.objects.
+                                       values_list('name', flat=True))
+            if not set(frameworks).issubset(available_frameworks):
+                raise ArgumentError('Unknown framework provided.')
+        if repositories:
+            if not set(repositories).issubset(set(Command.repos_to_retrigger_on)):
+                raise ArgumentError('Unknown repository provided.')
+
+    def _fetch_by(self, summaries_to_retrigger: QuerySet, frameworks: List[str], repositories: List[str]) -> QuerySet:
```

Comment:
Perhaps it'd be cleaner to separate out the functions that don't actually pertain to parsing/validating args, and have a new class or funcs called in `handle`.

Reply:
Ok. The Command class will only parse arguments and a new class will do the alert selection. The new class will be placed in alerts.py file.